src/main/msp/msp_serial.c

```diff
@@ -174,20 +165,20 @@ static mspPostProcessFuncPtr mspSerialProcessReceivedCommand(mspPort_t *msp)
     return mspPostProcessFn;
 }
 
-void mspSerialProcess(void)
+void mspSerialProcess(mspArmedState_e armedState)
 {
     for (uint8_t portIndex = 0; portIndex < MAX_MSP_PORT_COUNT; portIndex++) {
         mspPort_t * const mspPort = &mspPorts[portIndex];
         if (!mspPort->port) {
             continue;
         }
-        mspPostProcessFuncPtr mspPostProcessFn = NULL;
+        mspPostProcessFnPtr mspPostProcessFn = NULL;
         while (serialRxBytesWaiting(mspPort->port)) {
 
             const uint8_t c = serialRead(mspPort->port);
-            const bool consumed = mspProcessReceivedData(mspPort, c);
+            const bool consumed = mspSerialProcessReceivedData(mspPort, c);
 
-            if (!consumed && !ARMING_FLAG(ARMED)) {
+            if (!consumed && armedState == MSP_ARMED) {
```

Comment:
Passing a parameter rather than using a global state reducing coupling. It means the serial code is totally independent of anything else.

One of the motivations is to allow merge back into Cleanflight, where this code is used in the OSD board.